Release Notes 0.7 (#192)
kyma-project/website: src/blog-posts/13-release-notes-07.md

---
path: "/blog/release-notes-07"
date: "2019-02-08"
author: "Barbara Szwarc, Technical Writer @Kyma"
tags:
  - release-notes
title: "Kyma 0.7 Dublin"
---

> <a class="btn-blog" href=<><>Download 0.7</a>

Reviewer: Please change that line to `<a class="btn-blog" href="https://github.com/kyma-project/kyma/releases/tag/0.7.0" alt="Download 0.7">Download 0.7</a>`

## Asset Store

### Asset Store introduction

We have introduced the first version of a new, multi-cloud solution for storing assets such as documentation, images, or even client-side applications. <!-- Add a link to docs -->

This version of the Asset Store brings you the following features included in the [Asset Store proposal](https://github.com/kyma-project/community/blob/master/sig-and-wg/sig-core/proposals/asset-store-proposal.md):

* Fetching single files and packages (zip and tar files are supported for unpacking).
* Starting Minio in Minio Gateway mode to push resources to cloud providers like Azure or Google Cloud.
* Providing custom validation services for assets to validate them before uploading them to the storage.
* Providing custom mutation services for assets to modify them before uploading them to the storage. 

In the coming releases we plan to address the remaining features from the proposal. If you are interested in contributing, feel free to contact us.

## Console

### UI views migrated to SAP Fiori Fundamentals

> We have migrated Console UI views to SAP Fiori Fundamentals which uses [this](https://github.com/SAP/fundamental-react) library. In addition to the migration, we added the missing UI tests and improved the existing ones. This way, we covered all end-to-end scenarios for the Service Catalog views, and based them on the [testing bundle](https://github.com/kyma-project/bundles/tree/master/bundles/testing-0.0.1).

Reviewer: We have a core written in Angular, so we also use this library https://github.com/SAP/fundamental-ngx

### Upgrade to Kubernetes v1.11.5

You can now deploy Kyma using Kubernetes v1.11.5, to enjoy improved stability and new Kubernetes features such as the Status subresource for CRDs, which summarizes the current state of the object in the system.

### Kyma on Azure Kubernetes Service

You can now smoothly deploy Kyma on an Azure Kubernetes Service (AKS) cluster, using a chosen release version or even a particular commit. [Install](https://kyma-project.io/docs/master/root/kyma#installation-install-kyma-on-an-aks-cluster) Kyma on a cluster using a proprietary installer based on a Kubernetes operator. If you want to try out Kyma on a cluster without assigning the cluster to a domain you own, you can use xip.io which provides a wildcard DNS for any IP address. For details, see [this guide](https://kyma-project.io/docs/master/root/kyma#installation-install-kyma-on-an-aks-cluster-with-wildcard-dns).
 

### kyma-default ResourceQuota removed  

> You can install Kyma on Kubernetes clusters such as AKS or GKE which allow for autoscaling of nodes. Therefore, we removed the `kyma-default` ResourceQuota with a memory limit defined for resources. That change affects the `kyma-system`, `kyma-integration`, and `kyma-installer` Namespaces. Namespaces labelled with `env=true` still have the `kyma-default` ResourceQuota defined.

Reviewer 2: ```suggestion
You can install Kyma on Kubernetes clusters such as AKS or GKE which allow for nodes autoscaling. Therefore, we removed the `kyma-default` ResourceQuota with a memory limit defined for resources. That change affects the `kyma-system`, `kyma-integration`, and `kyma-installer` Namespaces. Namespaces labelled with `env=true` still have the `kyma-default` ResourceQuota defined.
```

We are constantly increasing the number of default alert rules, so you can receive alerts for affected components. The newest addition to the set is the alert rule for Persistent Volumes, activated when the volume used in the `kyma-system` Namespace exceeds 90%.

### Prometheus Operator now optional

The Prometheus Operator module used to be required in any Kyma setup. To increase flexibility, we moved the ServiceManager CRD from the Prometheus Operator to the CRD set provided by Kyma. This way, you can define Service Management resources for your components without the need to install the Prometheus Operator. 

## Security

###  UI API Layer security model 

We have switched from a custom Istio RBAC implementation based on Envoy to Kubernetes RBAC to avoid the difficulties of developing and maintainig the LUA filters for Envoy. Additionally, the current implementation uses Go which makes it more lightweight and flexible. Read the [documentation](docs/components/security#details-graphql) to learn the details on securing resources in GraphQL using queries, mutations, and subscriptions.

## Service Catalog

### Azure Service Broker runtime bundling and installation

> Kyma allows you to integrate with the Azure Service Broker in an easy way and to use the vast rage of Service Classes provided by this broker. All steps you need to perform to integrate with the Azure Service Broker are comprehensively described in the documentation, together with the description of each Service Class.

Reviewer 2: And that is one of the missing doc links, right?

### One-time provisioning for Helm Broker

> The Helm Broker now supports the `provisionOnlyOnce` defining whether the bundle should be provisioned only once for a given Namespace. We have also added the flag to Azure Service Broker and GCP Service Broker bundle configurations. For details, read [this](#docs/components/helm-broker#configuration-binding-bundles) document.

Reviewer 2: ```suggestion
The Helm Broker now supports the `provisionOnlyOnce` that defines whether the bundle should be provisioned only once for a given Namespace. We have also added the flag to the Azure Service Broker and GCP Service Broker bundle configurations. For details, read [this](#docs/components/helm-broker#configuration-binding-bundles) document.
```

## Service Mesh

### Default Istio sidecar injection 

> By default, the Istio sidecar injector watches all Pod creation operations on all Namespaces and it injects the newly created Pods with a sidecar proxy. To disable proxy injection for a Namespace, add the `istio-injection=disabled` label. To disable proxy injection on a Pod level, add the `sidecar.istio.io/inject:false` label to a Pod.

Reviewer 2: ```suggestion
By default, the Istio sidecar injector watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy. To disable proxy injection for a Namespace, add the `istio-injection=disabled` label. To disable proxy injection on a Pod level, add the `sidecar.istio.io/inject:false` label to a Pod.
```

---
path: "/blog/release-notes-07"
date: "2019-02-08"
author: "Barbara Szwarc, Technical Writer @Kyma"
tags:
  - release-notes
title: "Kyma 0.7 Dublin"
---

<a class="btn-blog" href="https://github.com/kyma-project/kyma/releases/tag/0.7.0" alt="Download 0.7">Download 0.7</a>

> The 0.7 release brings you features and improvements to further enhance your Kyma experience. The Asset Store, our new component, allows you to store assets ranging from documentation to customer applications. The ongoing CI development resulted in new pipelines which you can use to run Kyma installation and intercept any installation-related issues. The Kyma-Knative integration continues, making it possible to publish Events using Knative Eventing. Other significant improvements allow you to install Kyma on AKS, keep your resources secure with the new security model, and benefit from smoother and simplified Kyma deployment process.

Reviewer 3: `The Asset Store, our new component, allows you to store assets ranging from documentation to customer applications`
please change it, make sure it is stated that it is Alpha release, and the info about `customer applications` should not be there as in alpha version it doesn't make much sense to use it and I would not advertise it in first paragraph. So there are assets like images or markdown documentation

<!-- overview -->

The highlights of Kyma 0.7 Dublin include:

> - [Asset Store](#asset-store) - We introduced a brand new component for storing assets.

Reviewer 3: it is an alpha release

- [Continuous Integration](#continuous-integration) - New CI pipelines
- [Eventing](#eventing) - Publishing integrated with Knative Eventing
- [Installation](#installation) - Upgrade to Minikube v0.33.0, upgrade to Kubernetes v1.11.5, Kyma on Azure Kubernetes Service, resource quota removed from system Namespaces 
- [Monitoring](#monitoring) - New alert rule for Persistent Volumes, optional Prometheus Operator 
- [Security](#security) - UI API Layer security model 
- [Service Catalog](#service-catalog) - Azure Service Broker runtime bundling and installation, one-time provisioning for Helm Broker, Service Catalog standalone installation
- [Service Mesh](#service-mesh) - Default Istio sidecar injection, API Server Proxy autoscaling, startup scripts improvements
- [Tracing](#tracing) - Tracing UI elements renaming 

---

## Application Connector

### Increased Application Registry memory limit

We have increased the default memory limit for the Application Registry payload to 5 MB to enable registration of extensive API specifications.

### API secured with client certificates

> Application Proxy allows you to secure APIs with generated client certificates. When you register an API with a client certificate, the Application Registry generates a readu to ise certificate and key pair for this API to secure it. You can use the generated pair or replace it with your own certificate and key. For details, see [this](docs/0.7/components/application-connector#tutorials-register-a-secured-api) document.

Reviewer 2: ```suggestion
Application Proxy allows you to secure APIs with generated client certificates. When you register an API with a client certificate, the Application Registry generates a ready-to-use certificate and key pair for this API to secure it. You can use the generated pair or replace it with your own certificate and key. For details, see [this](docs/0.7/components/application-connector#tutorials-register-a-secured-api) document.
```

### Known issues

If you run Kyma on Minikube, you must expose the Nginx Ingress Controller on the dedicated NodePort. However, URLs returned from the Connector Service do not include the value for this port. To fix it, update URLs manually using the information in [this](docs/0.7/components/application-connector#api-application-registry) guide.

## Asset Store

### Asset Store introduction

We have introduced the first version of a new, multi-cloud solution for storing assets, such as documents, files, API specifications, or even client-side applications.

This version of the Asset Store brings you the following features included in the [Asset Store proposal](https://github.com/kyma-project/community/blob/master/sig-and-wg/sig-core/proposals/asset-store-proposal.md):

* Fetching single files and packages (zip and tar files are supported for unpacking).
* Starting Minio in the Minio Gateway mode to push resources to cloud providers, such as Azure or Google Cloud.
* Providing custom validation services for assets to validate them before uploading them to the storage.
* Providing custom mutation services for assets to modify them before uploading them to the storage. 

> In the coming releases, we plan to address the remaining features from the proposal. If you are interested in contributing, feel free to contact us. Meanwhile, read [this document](/docs/0.7/components/asset-store#overview-overview) to find out more about the Asset Store component.

Reviewer 3: please add here `feel free to contact us` that they can join us on slack, and add this link http://slack.kyma-project.io

## Console

### UI views migrated to SAP Fiori Fundamentals

> We have migrated Console UI views to SAP Fiori Fundamentals which uses [this](https://github.com/SAP/fundamental-ngx) library. In addition to the migration, we added the missing UI tests and improved the existing ones. This way, we covered all end-to-end scenarios for the Service Catalog views and based them on the [testing bundle](https://github.com/kyma-project/bundles/tree/master/bundles/testing-0.0.1).

Reviewer 3: In the input for RN I provided this link https://github.com/SAP/fundamental-react because the Catalog part of the Console UI is in React and we used that. If Angular views were also fully provided using https://github.com/SAP/fundamental-ngx  then we need to make it clear here in this sentence

## Eventing 

### Event Publishing with Knative Eventing

As a result of continuous Kyma-Knative integration, you can now deploy Kyma with a `--knative` feature flag. It allows you to use the Publish API provided by Knative Eventing.

## Installation

### Upgrade to Minikube v0.33.0 

We have upgraded Minikube to version [0.33.0](https://github.com/kubernetes/minikube/blob/master/CHANGELOG.md#version-0330---2019-01-17). This means we can now use the stable `kubeadm` bootstrapper instead of the deprecated `localkube`. This helps us to leverage new features and improves the stability of local installations.

### Upgrade to Kubernetes v1.11.5

You can now deploy Kyma using Kubernetes v1.11.5, to enjoy improved stability and new Kubernetes feature. One of them is the Status subresource for CRDs, which summarizes the current state of the object in the system.

### Kyma on AKS

> You can now smoothly deploy Kyma on an AKS cluster, using a chosen release version or even a particular commit. [Install](/docs/0.7/root/kyma#installation-install-kyma-on-an-aks-cluster) Kyma on a cluster using a proprietary installer based on a Kubernetes operator. If you want to try out Kyma on a cluster without assigning the cluster to a domain you own, you can use `xip.io` which provides a wildcard DNS for any IP address. For details, see [this guide](/docs/0.7/root/kyma#installation-install-kyma-on-an-aks-cluster-with-wildcard-dns).

Reviewer 4: anAks

## Security

###  UI API Layer security model 

We have switched from a custom Istio RBAC implementation based on Envoy to Kubernetes RBAC. The reason was to avoid difficulties of developing and maintaining the LUA filters for Envoy. Additionally, the current implementation uses Go, which makes it more lightweight and flexible. Read the [documentation](docs/0.7/components/security#details-graphql) to learn the details on securing resources in GraphQL using queries, mutations, and subscriptions.

## Service Catalog

### Azure Service Broker runtime bundling and installation

Kyma allows you to integrate with the Azure Service Broker in an easy way, and to use the vast range of Service Classes it provides. Each Service Class is comprehensively described in the [documentation](https://github.com/kyma-project/kyma/tree/master/docs/service-brokers). Read more about the Azure Service Broker [here](https://kyma-project.io/docs/master/components/helm-broker#overview-azure-broker).

### One-time provisioning for Helm Broker

> The Helm Broker now supports the `provisionOnlyOnce` flag which defines that the bundle should be provisioned only once for a given Namespace. We have also added this flag to the Azure Service Broker and the GCP Service Broker bundle configurations. For details, read [this](docs/0.7/components/helm-broker#configuration-binding-bundles) document.

Reviewer 4: supportsthe
This - link doesnt work

## Service Catalog standalone installation

> We took futher steps to modularize the Service Catalog by moving `service-catalog-addons` and `helm-broker` charts out of the `core` chart directory. This way, it's up to you to decide which charts to install.

Reviewer 5: ```suggestion
We took further steps to modularize the Service Catalog by moving `service-catalog-addons` and `helm-broker` charts out of the `core` chart directory. This way, it's up to you to decide which components to install. When other Kyma components will be moved out of the `core` chart, you'll be able to install Kyma with the Service Catalog only.

> We took futher steps to modularize the Service Catalog by moving `service-catalog-addons` and `helm-broker` charts out of the `core` chart directory. This way, it's up to you to decide which components to install. When other Kyma components will be moved out of the `core` chart, you will be able to install Kyma with the Service Catalog only.

Reviewer 2: ```suggestion
We took further steps to modularize the Service Catalog by moving `service-catalog-addons` and `helm-broker` charts out of the `core` chart directory. This way, it's up to you to decide which components to install. When other Kyma components will be moved out of the `core` chart, you will be able to install Kyma with the Service Catalog only.
```

## Service Mesh

### Default Istio sidecar injection 

By default, the Istio sidecar injector watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy. To disable proxy injection for a Namespace, add the `istio-injection=disabled` label. To disable proxy injection on a Pod level, add the `sidecar.istio.io/inject:false` label to a Pod.

### API Server Proxy autoscaling

To increase the availability of components, we have enabled the autoscaling of the API Server Proxy component. Based on the memory usage metric, the system decides whether you should deploy an additional instance of the API Server Proxy.

### Startup scripts improvements

To simplify the local deployments, the startup scripts now accept the `--password` argument. Running the scripts with this argument allows you to install Kyma locally and set a password for the admin user. This way, you don't need to fetch and decode the password generated for every deployment.  

## Tracing 

### Tracing UI elements renaming             

> We renamed the services and operations displayed in the Tracing UI to provide clarity and improved user experience.

Reviewer 5: ```suggestion
We renamed the services and operations displayed in the Tracing UI to provide clarity and better user experience.
```
